packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -709,26 +861,48 @@ class _RenderTheatre extends RenderBox with ContainerRenderObjectMixin<RenderBox
     addAll(children);
   }
 
-  bool _hasVisualOverflow = false;
+  @override
+  _RenderTheatre get theatre => this;
 
   @override
   void setupParentData(RenderBox child) {
-    if (child.parentData is! StackParentData) {
-      child.parentData = StackParentData();
+    if (child.parentData is! _TheatreParentData) {
+      child.parentData = _TheatreParentData();
+    }
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    RenderBox? child = firstChild;
+    while (child != null) {
+      final _TheatreParentData childParentData = child.parentData! as _TheatreParentData;
+      childParentData.topmostChildModel?.visitChildren((RenderObject child) => child.attach(owner));
+      child = childParentData.nextSibling;
     }
   }
 
-  Alignment? _resolvedAlignment;
+  static void _detachChild(RenderObject child) => child.detach();
 
-  void _resolve() {
-    if (_resolvedAlignment != null) {
-      return;
+  @override
+  void detach() {
+    super.detach();
```

Comment:
In the [docs]() it says

"Implementations of this method should end with a call to the inherited method, as in super.detach()."

But actually I think the docs may be wrong?  They also say:

"Subclasses with children should override this method to first call their inherited detach method, and then detach all their children."

If the docs are wrong I'm happy to update that or at least make it more clear.

Reply:
The link points to this PR. But imo it's better to call `super.detach` first to prevent the parent from getting into its owner's dirty list.